doc/src/build/package-versioning.md

```diff
@@ -0,0 +1,30 @@
+---
+title: Package Versioning
+---
+
+Sui smart contracts are organized into [upgradeable](./package-upgrades.md) packages and, as a result, multiple versions of any given package can exist on-chain. Before a package can be used on the chain, it is [published](move/debug-publish.md#publishing-a-package) at its first (a.k.a, original) version. A new version of a package is created whenever the package is [upgraded](./package-upgrades.md), and an upgrade always happens with respect to the version of the package immediately preceding the upgraded one in the versions history. In other words, an upgrade to the Nth version of a package is always based on the N-1th version of the package (this is enforced by the system) and one can never create the Nth version of the package based on a version that's smaller than N-1th version. For example a package can be upgraded from version 1 to 2 but once this happens only an upgrade from version 2 to 3 is possible and an upgrade from version 1 to 3 is forbidden.
+
+Note that there is also a notion of versioning present in the [manifest](./move/manifest.md) files. You can observe it both in the [package section](./move/manifest.md#package-section) and in the [dependencies section](./move/manifest.md#dependencies-section). For example:
+```move
+[package]
+name = "some_pkg"
+version = "1.0.0"
+
+[dependencies]
+another_pkg = { git = "https://github.com/another_pkg/another_pkg.git" , version = "2.0.0"}
+```
+These fields in the manifest file, however, at this point are only used for user-level documentation purposes and are ignored by the build or publish/upgrade commands. Another way of looking at it is that if a developer publishes a package with a certain package version in the manifest file and then modifies and re-publishes the same package with a different version (using publish command rather than upgrade command), these two packages will be considered different packages, rather than on-chain versions of the same package, and recognized as such by the tooling - for example, none of these packages could be used as a [dependency override](./dependency-overrides.md) to stand in for the other one.
```

Comment:
This is not true today, AFAICT, if you publish, change, publish, then the (off-chain) tooling will allow you to use the second package as an override of the first, but things will go wrong when you try and publish on-chain.  Maybe it's this second part that you're refering to, in which case we should clarify that.

Reply:
It is indeed the second part that I meant and tried to skim over the details a bit but I agree that it might have been misleading a bit. I tried rephrasing to reflect what's going on a bit better.